src/frontend/src/pages/FinancePage/CreateReimbursementRequestForm/ReimbursementProductTable.tsx

```diff
@@ -0,0 +1,167 @@
+                </TableCell>
+                <TableCell>
+                  <Box sx={{ display: 'flex', flexWrap: 'wrap', listStyle: 'none', p: 0.5, m: 0 }} component={'ul'}>
+                    {uniqueWbsElementsWithProducts.get(key)?.map((product, index) => {
+                      return product.name.length > 0 && product.cost > 0 ? (
+                        <ListItem key={product.index}>
+                          <Chip label={`${product.name} - $${product.cost}`} onDelete={() => removeProduct(index)} />
+                        </ListItem>
+                      ) : (
+                        <ListItem key={product.index}>
+                          <Grid container spacing={1}>
+                            <Grid item md={6} xs={12}>
+                              <Controller
+                                name={`reimbursementProducts.${product.index}.name`}
+                                control={control}
+                                render={({ field }) => (
+                                  <TextField {...field} label={'Description'} size={'small'} variant={'outlined'} />
+                                )}
+                              />
+                            </Grid>
+                            <Grid item md={6} xs={12} display={'flex'}>
+                              <Controller
+                                name={`reimbursementProducts.${product.index}.cost`}
+                                control={control}
+                                render={({ field }) => (
+                                  <TextField
+                                    {...field}
+                                    label={'Cost'}
+                                    size={'small'}
+                                    variant={'outlined'}
+                                    type="number"
+                                    InputProps={{
+                                      startAdornment: <InputAdornment position="start">$</InputAdornment>
+                                    }}
+                                  />
+                                )}
+                              />
+                              <IconButton onClick={() => removeProduct(product.index)}>
+                                <Delete />
+                              </IconButton>
+                            </Grid>
+                          </Grid>
+                        </ListItem>
+                      );
+                    })}
+                  </Box>
+                </TableCell>
+                <TableCell>
+                  <IconButton
+                    onClick={() =>
+                      appendProduct({
+                        wbsNum: validateWBS(key),
+                        name: '',
+                        cost: 0
+                      })
+                    }
+                  >
+                    <Add />
+                  </IconButton>
```

Comment:
I think this should be aligned with the bottom product on the wbsElement, looks weird when there are multiple products rn

Reply:
really instead of centered?

Reply:
give it a try, I think just where the add button is right now is kinda weird; if it looks better centered you can leave it that way